api/ical_api.py

```diff
@@ -0,0 +1,187 @@
+
+    calendar_url = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ApplicationEvent
+        fields = ["calendar_url"]
+
+    def get_calendar_url(self, instance):
+        request = self.context["request"] if "request" in self.context else None
+        scheme = (
+            urlsplit(request.build_absolute_uri(None)).scheme if request else "http"
+        )
+        host = (
+            request.META["HTTP_HOST"]
+            if request and hasattr(request.META, "HTTP_HOST")
+            else None
+        )
+        calendar_url = reverse(
+            "reservation_unit_calendar-detail", kwargs={"pk": instance.id}
+        )
+
+        return f"{scheme}://{host}{calendar_url}?hash={uuid_to_hmac_signature(uuid=instance.uuid)}"
+
+
+class ReservationUnitCalendarUrlViewSet(viewsets.ModelViewSet):
+    serializer_class = ReservationUnitCalendarUrlSerializer
+    queryset = ReservationUnit.objects.all()
+    permission_classes = (
+        [ReservationUnitCalendarUrlPermission]
+        if not settings.TMP_PERMISSIONS_DISABLED
+        else [permissions.AllowAny]
+    )
+
+
+class ReservationUnitIcalViewset(ViewSet):
+    queryset = ReservationUnit.objects.all()
+
+    def get_object(self):
+        return ReservationUnit.objects.get(pk=self.kwargs["pk"])
+
+    def retrieve(self, request, *args, **kwargs):
+        hash = request.query_params.get("hash", None)
+        if not hash:
+            raise ValidationError("hash is required")
+
+        instance = self.get_object()
+
+        comparison_signature = uuid_to_hmac_signature(uuid=instance.uuid)
+
+        if not hmac.compare_digest(comparison_signature, hash):
+            raise ValidationError("invalid hash signature")
+
+        host = (
+            request.META["HTTP_HOST"]
+            if request and hasattr(request.META, "HTTP_HOST")
+            else None
+        )
+
+        buffer = io.BytesIO()
+        buffer.write(
+            export_reservation_unit_events(
+                instance, host, reservation_unit_calendar(instance)
+            ).to_ical()
+        )
+        buffer.seek(0)
+
+        return FileResponse(buffer, as_attachment=True, filename="file.ics")
+
+
+class ApplicationEventIcalViewset(ViewSet):
+    queryset = ApplicationEvent.objects.all()
+
+    def get_object(self):
+        return ApplicationEvent.objects.get(uuid=self.kwargs["pk"])
+
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        buffer = io.BytesIO()
+        buffer.write(
+            export(
+                instance,
+                request.META["HTTP_HOST"],
+                application_event_calendar(instance),
+            ).to_ical()
+        )
+        buffer.seek(0)
+
+        return FileResponse(buffer, as_attachment=True, filename="file.ics")
+
+
+def reservation_unit_calendar(reservation_unit: ReservationUnit):
+    cal = Calendar()
+
+    cal.add("version", "2.0")
```

Comment:
should the version come from some variable / settings?

Reply:
I guess it could be good.

Reply:
Setting might be a bit heavy handed, maybe just a constant would be better.